### ISSUE TITLE
Nodes restart continuously when adding plugin.

### ISSUE THREAD

**Issue by vipinjn24** [Aiven-Open/prometheus-exporter-plugin-for-opensearch#248]
Using OpenSearch 2.11.1

After adding the plugin in k8s-operator, it seems that the cluster nodes goes to restart all the time, and never gets stable.
I confirm that before adding the plugin the nodes were working fine for the same release.

please see attached log of one of the master nodes.
[opensearch.log](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/files/14138554/opensearch.log)

This issue is in 2.11.0 also, i tried 2.8.0 and it works fine, but not these 2.

Cluster is baremetal

**Comment by lukas-vlcek** [comment]
Edit: I removed the release content from the ticket description.

I do not see anything suspicious in the log. Is the log complete?

It might be either the plugin or the k8s operator... hard to say right now. In any case it seems that having an integration test with the k8s operator will be useful, something relevant to https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/issues/240

**Comment by lukas-vlcek** [comment]
@vipinjn24 One more note, the test suit for this plugin contains [tests](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/tree/main/src/yamlRestTest/resources/rest-api-spec/test) that do full cluster formation with the plugin installed on both nodes. The cluster consists of [two nodes](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/blob/main/build.gradle#L128). Yes, it does **not** include the security plugin but at least a basic smoke test is part of every release. If you can provide more complete logs that would be great.

**Comment by vipinjn24** [comment]
[opensearch-node.log](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/files/14169080/opensearch-node.log)
[opensearch-coordinator.log](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/files/14169081/opensearch-coordinator.log)
[opensearch-master.log](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/files/14169082/opensearch-master.log)

These are attached files.

only 2 master nodes bootstrapped after 7 restarts.
1 master 2 data nodes and 1 coordinator node still doing restarts with different logs

![image](https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/assets/1639185/8c271bb8-7d95-4de9-b7e8-a9a9f1eebab1)

**Comment by vipinjn24** [comment]
IDK, this time i deleted the cluster and created it from scratch and now it works, strange :(

**Comment by lukas-vlcek** [comment]
Maybe there was something wrong/corrupted with the data stored on persistent volumes if anything like that was re-attached to the nodes?

**Comment by vipinjn24** [comment]
I am 100% sure i deleted it before hand. but the cluster was still restarting.

**Comment by vipinjn24** [comment]
hmm now i restarted the kubernetes cluster and the entire cluster fails to start, trying to restart after removing the plugin

**Comment by vipinjn24** [comment]
I did some more digging, and found that this restarts are related to the startup probes added to the nodes.
Since these probes are hard coded, I had to raise a PR to update the logic of the operator and the charts.
[Pull Request](https://github.com/opensearch-project/opensearch-k8s-operator/pull/728)

I was able to build this code locally and add to my network docker registry and updated the operator to use this image. this is now working fine after updating the failure threshold to somewhat more than what was initially 10.

Hoping to soon get this merged and released officially.

We can mark this issue closed
